src/frontend/sharedComponents/MenuContent.tsx

```diff
@@ -0,0 +1,287 @@
+import {useIntl, defineMessages} from 'react-intl';
+import IonIcon from 'react-native-vector-icons/Ionicons';
+import Octicons from 'react-native-vector-icons/Octicons';
+import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
+
+import DeviceIcon from '../images/DeviceIcon.svg';
+import Exchange from '../images/Exchange.svg';
+import {CloseIcon} from './icons';
+
+import {MenuListItem} from './MenuList/MenuListItem';
+import {HeaderText} from './Text/HeaderText';
+import {BodyText} from './Text/BodyText';
+import {PrimaryButton, SecondaryButton} from './Buttons';
+import {Divider} from './Divider';
+
+import {useNavigationFromRoot} from '../hooks/useNavigationWithTypes';
+import {useOwnDeviceInfo} from '@comapeo/core-react';
+import {useProjectRole} from '../hooks/useProjectRole';
+import {useProjectSettings} from '../hooks/server/projects';
+
+import {
+  NEW_DARK_GREY,
+  WHITE,
+  BLUE_GREY,
+  VERY_LIGHT_GREY,
+  BLACK,
+} from '../lib/styles';
+
+const m = defineMessages({
+  aboutCoMapeo: {
+    id: 'Navigation.Menu.aboutCoMapeo',
+    defaultMessage: 'About CoMapeo',
+  },
+  appSettings: {
+    id: 'Navigation.Menu.Settings',
+    defaultMessage: 'Settings',
+  },
+  privacyPolicy: {
+    id: 'Navigation.Menu.privacyPolicy',
+    defaultMessage: 'Data & Privacy',
+  },
+  mappingOnOwn: {
+    id: 'Navigation.Menu.mappingOnOwn',
+    defaultMessage: 'You are mapping on your own.',
+  },
+  coordinator: {
+    id: 'Navigation.Menu.coordinator',
+    defaultMessage: 'You are a coordinator on this project.',
+  },
+  participant: {
+    id: 'Navigation.Menu.participant',
+    defaultMessage: 'You are a participant on this project.',
+  },
+  currentProject: {
+    id: 'Navigation.Menu.currentProject',
+    defaultMessage: 'Current Project',
+  },
+  exchange: {
+    id: 'Navigation.Menu.exchange',
+    defaultMessage: 'Exchange',
+  },
+  invite: {
+    id: 'Navigation.Menu.invite',
+    defaultMessage: 'Invite',
+  },
+  viewProject: {
+    id: 'Navigation.Menu.viewProject',
+    defaultMessage: 'View',
+  },
+});
+
+type MenuContentProps = {
+  closeMenu: () => void;
+};
+
+export function MenuContent({closeMenu}: MenuContentProps) {
+  const {data: deviceData} = useOwnDeviceInfo();
+  const {data: projectData} = useProjectSettings();
+  const role = useProjectRole(projectData);
+  const {formatMessage} = useIntl();
+  const {navigate} = useNavigationFromRoot();
+
+  const deviceName = deviceData?.name;
+  const projectName = projectData?.name || 'My Solo Project';
+
+  return (
+    <View style={styles.container}>
+      <View style={styles.topBar}>
+        <View style={styles.deviceRow}>
+          <DeviceIcon width={32} height={32} />
+          <HeaderText variant="header4">{deviceName}</HeaderText>
+        </View>
+        <TouchableOpacity onPress={closeMenu} accessibilityLabel="Close Menu">
+          <CloseIcon size={32} />
+        </TouchableOpacity>
+      </View>
```

Comment:
I think we should be leveraging React navigations header to be consistent. So creating this header as its own component and setting it at the `Stack/AppScreens.tsx` level